src/AEG/AEGUtils.ts

```diff
@@ -70,25 +70,72 @@ export function shapesIntersect(
  * @returns True, if the shapes intersect. Else, false.
  */
 function ellipseRectangleIntersection(ellipse: Ellipse, rectangle: Rectangle): boolean {
-    //For ellipse-rectangle collision, check if there are corners of the rectangle both within
-    //and outside the ellipse
+    //For ellipse-rectangle collision, check for intersection along the horizontal or vertical edges
     const corners = rectangle.getCorners();
-    let cornerInEllipse = false;
-    let cornerOutOfEllipse = false;
-    let val: number;
+    const wideCoords = getWidestCoordinates(ellipse);
 
-    for (let i = 0; i < 4; i++) {
-        val = signedDistanceFromEllipse(ellipse, corners[i]);
-        if (val < 0) {
-            //The corner is within the ellipse
-            cornerInEllipse = true;
-        } else if (val > 0) {
-            //The corner is outside the ellipse
-            cornerOutOfEllipse = true;
-        }
+    let leftBound: number;
+    let rightBound: number;
+    let topBound: number;
+    let bottomBound: number;
+
+    if (ellipse.radiusX >= rectangle.width) {
+        leftBound = wideCoords[3].x;
+        rightBound = wideCoords[1].x;
+    } else {
+        leftBound = corners[0].x;
+        rightBound = corners[1].x;
     }
 
-    return cornerInEllipse && cornerOutOfEllipse;
+    if (ellipse.radiusY >= rectangle.height) {
+        topBound = wideCoords[0].y;
+        bottomBound = wideCoords[2].y;
+    } else {
+        topBound = corners[0].y;
+        bottomBound = corners[2].y;
+    }
+
+    //Check for vertical edge intersection
+    const leftEdgeUpperCurve = getCurvePoint(ellipse, corners[0].x, 1);
+    const rightEdgeUpperCurve = getCurvePoint(ellipse, corners[1].x, 1);
+    const leftEdgeLowerCurve = getCurvePoint(ellipse, corners[0].x, -1);
+    const rightEdgeLowerCurve = getCurvePoint(ellipse, corners[1].x, -1);
+
+    const verticalIntersection =
+        (topBound < leftEdgeUpperCurve && leftEdgeUpperCurve < bottomBound) ||
+        (topBound < leftEdgeLowerCurve && leftEdgeLowerCurve < bottomBound) ||
+        (topBound < rightEdgeUpperCurve && rightEdgeUpperCurve < bottomBound) ||
+        (topBound < rightEdgeLowerCurve && rightEdgeLowerCurve < bottomBound);
+
```

Comment:
You should return early here if there is a vertical intersection, no need to do more math than we have to.